src/gui/MainWindow.cc

```diff
@@ -2952,56 +2962,43 @@ void MainWindow::actionExport(FileFormat format, const char *type_name, const ch
   clearCurrentOutput();
 }
 
-void MainWindow::actionExportSTL()
-{
-  if (Settings::Settings::exportUseAsciiSTL.value()) {
-    actionExport(FileFormat::ASCIISTL, "ASCIISTL", ".stl", 3);
-  } else {
-    actionExport(FileFormat::STL, "STL", ".stl", 3);
-  }
-}
-
-void MainWindow::actionExport3MF()
-{
-  actionExport(FileFormat::_3MF, "3MF", ".3mf", 3);
-}
-
-void MainWindow::actionExportOBJ()
-{
-  actionExport(FileFormat::OBJ, "OBJ", ".obj", 3);
-}
-
-void MainWindow::actionExportOFF()
-{
-  actionExport(FileFormat::OFF, "OFF", ".off", 3);
-}
-
-void MainWindow::actionExportWRL()
-{
-  actionExport(FileFormat::WRL, "WRL", ".wrl", 3);
-}
-
-void MainWindow::actionExportPOV()
-{
-  actionExport(FileFormat::POV, "POV", ".pov", 3);
-}
-
-void MainWindow::actionExportAMF()
-{
-  actionExport(FileFormat::AMF, "AMF", ".amf", 3);
-}
-
-void MainWindow::actionExportDXF()
-{
-  actionExport(FileFormat::DXF, "DXF", ".dxf", 2);
-}
-
-void MainWindow::actionExportSVG()
+void MainWindow::actionExportFileFormat(int fmt_)
 {
-  actionExport(FileFormat::SVG, "SVG", ".svg", 2);
-}
-
-void MainWindow::actionExportPDF()
+  FileFormat fmt = (FileFormat) fmt_;	
+  printf("Export Fileformat	\n");
+  switch(fmt) {
+    case FileFormat::STL:
+      if (Settings::Settings::exportUseAsciiSTL.value()) {
+        actionExport(FileFormat::ASCIISTL, "ASCIISTL", ".stl", 3);
+      } else {
+        actionExport(FileFormat::STL, "STL", ".stl", 3);
+      }
+      break;	    
+    case FileFormat::_3MF:
+      actionExport(FileFormat::_3MF, "3MF", ".3mf", 3);
+      break;
+    case FileFormat::OBJ:
+      actionExport(FileFormat::OBJ, "OBJ", ".obj", 3);
+      break;
+    case FileFormat::OFF:
+      actionExport(FileFormat::OFF, "OFF", ".off", 3);
+      break;
+    case FileFormat::WRL:
+      actionExport(FileFormat::WRL, "WRL", ".wrl", 3);
+      break;
+    case FileFormat::POV:
+      actionExport(FileFormat::POV, "POV", ".pov", 3);
+      break;
+    case FileFormat::AMF:
+      actionExport(FileFormat::AMF, "AMF", ".amf", 3);
+      break;
+    case FileFormat::DXF:
+      actionExport(FileFormat::DXF, "DXF", ".dxf", 2);
+      break;
+    case FileFormat::SVG:
+      actionExport(FileFormat::SVG, "SVG", ".svg", 2);
+      break;
```

Comment:
I wonder if you could use `FileFormatInfo` to express this directly: `FileFormatInfo info = fileformat::info(fmt)`

Reply:
Cannot find any FileFormatInfo in all code. Did it recently disappear or do you rather refer to  ExportFileFormatOptions ?

Reply:
It was recently introduced. I wrote it a while ago, but I guess it only made it into master yesterday:
https://github.com/openscad/openscad/blob/11fb64f013b494399358dc4a5ff5de84847983b0/src/io/export.h#L41